Xamarin.Forms.Core/Layout.cs

```diff
@@ -319,69 +319,85 @@ internal static void LayoutChildIntoBoundingRegion(View child, Rectangle region,
+					{
+						return;
+					}
+					if (trigger == InvalidationTrigger.HorizontalOptionsChanged || trigger == InvalidationTrigger.VerticalOptionsChanged)
+					{
+						ComputeConstraintForView(view);
+					}
 				}
-			}
-
-			_allocatedFlag = false;
-			if (trigger == InvalidationTrigger.RendererReady)
-			{
-				InvalidateMeasureInternal(InvalidationTrigger.RendererReady);
-			}
-			else
-			{
-				InvalidateMeasureInternal(InvalidationTrigger.MeasureChanged);
-			}
-
-			s_resolutionList.Add(new KeyValuePair<Layout, int>(this, GetElementDepth(this)));
-			if (!s_relayoutInProgress)
-			{
-				s_relayoutInProgress = true;
 
-				// Rather than recomputing the layout for each change as it happens, we accumulate them in 
-				// s_resolutionList and schedule a single layout update operation to handle them all at once.
-				// This avoids a lot of unnecessary layout operations if something is triggering many property
-				// changes at once (e.g., a BindingContext change)
+				_allocatedFlag = false;
+				if (trigger == InvalidationTrigger.RendererReady)
+				{
+					InvalidateMeasureInternal(InvalidationTrigger.RendererReady);
+				}
+				else
+				{
+					InvalidateMeasureInternal(InvalidationTrigger.MeasureChanged);
+				}
 
-				Device.BeginInvokeOnMainThread(() =>
+				s_resolutionList.Add(new KeyValuePair<Layout, int>(this, GetElementDepth(this)));
+				if (!s_relayoutInProgress)
 				{
-					// if thread safety mattered we would need to lock this and compareexchange above
-					IList<KeyValuePair<Layout, int>> copy = s_resolutionList;
-					s_resolutionList = new List<KeyValuePair<Layout, int>>();
-					s_relayoutInProgress = false;
+					s_relayoutInProgress = true;
 
-					foreach (KeyValuePair<Layout, int> kvp in copy.OrderBy(kvp => kvp.Value))
+					// Rather than recomputing the layout for each change as it happens, we accumulate them in 
+					// s_resolutionList and schedule a single layout update operation to handle them all at once.
+					// This avoids a lot of unnecessary layout operations if something is triggering many property
+					// changes at once (e.g., a BindingContext change)
+
+					if (Dispatcher != null && Dispatcher.IsInvokeRequired)
+					{
+						Dispatcher.BeginInvokeOnMainThread(ResolveLayoutChanges);
+					}
+					else if(Device.IsInvokeRequired)
+					{
+						Device.BeginInvokeOnMainThread(ResolveLayoutChanges);
+					}			
+					else
 					{
-						Layout layout = kvp.Key;
-						double width = layout.Width, height = layout.Height;
-						if (!layout._allocatedFlag && width >= 0 && height >= 0)
-						{
-							layout.SizeAllocated(width, height);
-						}
+						ResolveLayoutChanges();
 					}
-				});
+				}
+			}
+		}
+
+		internal void ResolveLayoutChanges()
+		{
+			// if thread safety mattered we would need to lock this and compareexchange above
+			var copy = s_resolutionList.OrderBy(kvp => kvp.Value);
+			s_relayoutInProgress = false;
```

Comment:
This is missing the following line from the previous implementation:
`s_resolutionList = new List<KeyValuePair<Layout, int>>();`

Without that line, `s_resolutionList` will hold references to every element that ever gets a layout pass, so they can never be garbage-collected.